### PR TITLE
Fix `LATEST` version in KPM to point to even numbers #49

### PR DESCRIPTION
**Pull request by wwjbatista**

### CODE REVIEW

File: kpm/lib/kpm/kaui_artifact.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def versions(overrides={}, ssl_verify=true)
         coordinates = KPM::Coordinates.build_coordinates(coordinate_map)
         response    = REXML::Document.new nexus_remote(overrides, ssl_verify).search_for_artifacts(coordinates)
         versions    = SortedSet.new
-        response.elements.each('search-results/data/artifact/version') { |element| versions << element.text }
+        response.elements.each('searchNGResponse/data/artifact/version') { |element| versions << element.text }
```
Comment (pierre) [inline kpm/lib/kpm/kaui_artifact.rb]: Why this change?

File: kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def message
     class NexusApiCallsV2
       PULL_ARTIFACT_ENDPOINT = '/service/local/artifact/maven/redirect'
       GET_ARTIFACT_INFO_ENDPOINT = '/service/local/artifact/maven/resolve'
-      SEARCH_FOR_ARTIFACT_ENDPOINT = '/service/local/data_index'
+      SEARCH_FOR_ARTIFACT_ENDPOINT = '/service/local/lucene/search'
```
Comment (pierre) [inline kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb]: Why this change?

File: kpm/spec/kpm/unit/installer_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@
     it 'finds the right stable versions' do
       config = KPM::Installer.build_default_config(all_kb_versions)
       config['killbill'].should_not be_nil
-      config['killbill']['version'].should == '0.16.11'
+      config['killbill']['version'].should == '0.16.10'
```
Comment (pierre) [inline kpm/spec/kpm/unit/installer_spec.rb]: 0.16.11 is correct (latest one on the 0.16.x release train).

File: kpm/lib/kpm/installer.rb [hunk 4]
```diff
@@ -27,13 +34,12 @@ def self.build_default_config(all_kb_versions=nil)
         next if version.nil?
 
         major, minor, patch, pre = version.segments
-        next if !pre.nil? || minor.nil? || minor.to_i.odd?
+        next if !pre.nil? || minor.nil? || minor.to_i.odd? || patch.to_i.odd?
```
Comment (pierre) [inline kpm/lib/kpm/installer.rb]: No need to look at the patch version (0.18.11 is stable for instance -- see description at http://killbill.io/downloads/).

File: kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb (hunk 2 above)
Comment (wwjbatista) [inline kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb]: The "/data_index" does not return the complete list of versions at the moment of search, for example the latest version for Killbill with the endpoint "/data_index" gave me 0.18.6 and with "/lucene/search" gave me 0.18.12. Also the mount point "/data_index" is deprecated.

File: kpm/lib/kpm/kaui_artifact.rb (hunk 1 above)
Comment (wwjbatista) [inline kpm/lib/kpm/kaui_artifact.rb]: Since the endpoint was changed to "/lucene/search" the result search must be changed because the root element of the document returned is "searchNGResponse".  The "search-results" is for the endpoint "/data_index".

File: kpm/lib/kpm/installer.rb (hunk 4 above)
Comment (wwjbatista) [inline kpm/lib/kpm/installer.rb]: I was confused. I was in the assumption that only even numbers formed a stable version.

File: kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb (hunk 2 above)
Comment (pierre) [inline kpm/lib/kpm/nexus_helper/nexus_api_calls_v2.rb]: Ah interesting, I didn't know! 👍
